2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+
+  >Trainee provides preferred time for discussion session (optional)
+
+  * The *admin* fixes up a checkout session
+
+  >Trainees attend checkout session
+
+  * The *admin* evaluates the checkout session, providing necessary feedback. If satisfied, updates the checkout complete status
+  * When this is done, the applicant now becomes a trainee, and they can be managed from the AMY instructor dashboard.
+
+
+### 3. Trainee Dashboard
+
+  Every applicant that gets selected will be provided with an access to the trainee dashboard. This will be a one stop solution to all their queries. Most importantly, they'll know what has been done and what to do next. Here, the trainee can track their progress, interact with instructors at any stage, and find out the next steps need to be done. This makes things simpler for everyone.
+
+  * **Know alloted course**
+
+  As soon as the trainee logs in, they'll be shown with the course they are matched with. When they complete the course, they'll be taken to the next step - homework submission.
+
+  * **Homework Submission**
+
+  Trainees can submit the link to their pull request or Google Doc. This way it'll be easier to track submissions.
+
+  * **Discussion sessions**
+
+  The trainees will have an option to select a time slot out of the available ones. This makes matching easier.
+
+  * **Checkout session**
+
+  If the trainee passes the discussion session, they'll again be able to give preferred choices for checkout sessions.
+
+  * **Training complete**
+
+  Once the training is complete, the trainees can download certificates, and move to the AMY instructor dashboard
+
+### Salient Features
+
+  **In house messaging and feedback**
+
+  At any point of time, the trainee and the relevant instructors will be able to communicate from within the portal. This solves the hassle of unorganised emails.
+
+  **Preferrential selection of discussion and checkout sessions**
+
+  This makes the process flexible, and comfortable.
+
+  **Email notifications**
+
+  Email notifications would be sent in case of important events such as an upcoming discussion session, so that the applicant doesn't miss any deadlines
+
+### Flowchart
+
+Here's a graphic that summarizes the process
+
+![Image](https://docs.google.com/drawings/d/11PIidJrIwYlGbbb5PvyWXpa6r8lG0OVQ7ZJGbefU6VU/pub?w=920&h=1006)
+
+
+## Technical Details
+
+**Backend**
+
+The portal will use **Django** for obvious reasons.
+
+**Database**
+
+As the portal would be a subapp within the AMY project, it will use the same database as AMY. But since the portal, and AMY, in general would be handling a higher load, `PostgreSQL` would be the preferred database. An issue is already raised [here](https://github.com/swcarpentry/amy/issues/716). If this gets addressed, we'll use PostgreSQL. If not, the system can also run on SQLite during development phase.
```

Comment:
What new classes (if any) will you add to the models? How will they relate to existing model classes?